spec/component/models/user_profile_spec.rb

```diff
@@ -2,7 +2,7 @@
 
 require 'component/component_spec_helper'
 
-describe UserProfile do
+describe UserProfile, vcr: true do
```

Comment:
Instead of generating all of these fixtures, what does everybody think about just mocking `PsuIdentityUserService` for this whole spec? Then if we want to test the integration from `UserProfile` all the way up to the external API call to the PSU identity service, we can write just one test to do that specifically and stick it in spec/integration. Or... if we can trust that the psu_identity gem is doing its job right, then maybe our integration test doesn't need to mock the HTTP requests at all. It can just mock `PsuIdentity::SearchService::Client`.

Even if we do generally trust the `PsuIdentity::SearchService::Client` API wrapper, I think there's definitely value in having a single integration test somewhere that exercises the interaction with the real external service. That's generally how I approach testing against external APIs when a 3rd-party  wrapper is available and we've hidden the wrapper behind our own interface (I know that you're the "3rd party" in this case, but the point is that someone has already abstracted the HTTP calls):

1. mock our own interface (`PsuIdentityUserService` in this case) for all of the unit tests
2. maybe write a few component tests that exercise the interaction between our code and the 3rd-party wrapper while mocking the wrapper (`PsuIdentity::SearchService::Client` in this case), and of course test our own interface also while mocking the wrapper
3. then, if it's feasible to test against the real external service (or an identical sandbox), write one heavy-weight integration test that exercises the whole stack IRL

I wouldn't normally write any tests that use fixtures to mock HTTP requests and responses unless I wrote the code that was making the HTTP requests.

We probably wouldn't want a full-stack integration test to run as part of our default test suite in development, but we could tag it so that it only runs on CI, and accept the tradeoff of knowing that it's going to fail occasionally due of an outage for the assurance that our app actually still works with the real PSU identity service.

Reply:
I tend to agree with this, If we keep a single integration test that confirms the shape of the response from the identity api, we can mock out the rest. I tend to get wheezy with vcr, but that can also be chalked up to my limited understanding of vcr, and not a real problem.

Reply:
I can see a use for a tool like VCR (though I've never used it myself), but that use is really limited. I looks to me like it's easy to overuse it an a way that just ends up enabling anti-patterns.

Anyway, if it would be helpful (and if I get a chance), I could submit a separate PR to demonstrate what I'm suggesting here.

Reply:
Yeah I agree.  I can take a stab at this today.  `UserProfile` basically uses `PsuIdentity::SearchService::Client` every time it's instantiated, so it's sporadically sending requests throughout the test suite.